Production_scripts/B06_Process_PIS_extract.R

```diff
@@ -0,0 +1,251 @@
+#####################################################
+# Prescribing Extract
+# Author: Catherine Holland
+# Date: February 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input -
+# Description - Process Prescribing Extract
+#####################################################
+
+library(dplyr)
+library(createslf)
+library(slfhelper)
+
+
+# Read in data ---------------------------------------
+
+# latest year
+latest_year <- 1920
+
+pis_extract <- readr::read_csv(
+  get_it_prescribing_path(latest_year)
+) %>%
+  # rename
+  rename(
+    chi = "Pat UPI [C]",
+    dob = "Pat DoB [C]",
+    gender = "Pat Gender",
+    postcode = "Pat Postcode [C]",
+    gpprac = "Practice Code",
+    no_dispensed_items = "Number of Dispensed Items",
+    cost_total_net = "DI Paid NIC excl. BB"
+  ) %>%
+  # de-select "DI Paid GIC excl. BB"
+  select(-c(`DI Paid GIC excl. BB`)) %>%
+  # filter for chi NA
+  filter(!is.na(chi)) %>%
+  # create variables recid and year
+  mutate(
+    recid = "PIS",
+    year = latest_year
+  )
+
+
+# Recode GP Practice into a 5 digit number ---------------------------------------
+
+# assume that if it starts with a letter it's an English practice and so recode to 99995
+pis_extract <-
+  pis_extract %>%
+  mutate(gpprac = replace(gpprac, substr(gpprac, 1, 1) %in% c("A", "Z"), "99995"))
```

Comment:
This can be replaced with the `eng_gp_to_dummy` function.
```suggestion
pis_extract <-
  pis_extract %>%
  eng_gp_to_dummy(gpprac)
```

Reply:
@Moohan function `eng_gp_to_dummy` not working?

Reply:
fixed with `as.character`